File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

void draw(char c, int n);

int main(void)
{
    int n;

    //read the heigth of pyramid 1<=x<=8
    do
    {
        n = get_int("Heigh: ");
    }
    while (n < 1 || n > 8);

    //draw the pyramids
    for (int i = 1; i <= n; i++)
    {
        draw(' ', n - i);
        draw('#', i);
        draw(' ', 2);
        draw('#', i);
        printf("\n");
    }
}

void draw(char c, int n)
{
    for (int i = 1; i <= n ; i++)
    {
        printf("%c", c);
    }
}

File: pset6/mario/mario.py
from cs50 import get_int


def main():
    
    # prompts the user to asingn the heigth
    while True:
        height = get_int("Heigth:")
        if height > 0 and height <= 8:
            break
    
    # call a function to draw, passing the spaces and briks
    for i in range(height):
        print_line(height - i - 1, i + 1)


# function to print
def print_line(space, brick):
    print(" " * space, end="")
    print("#" * brick, end="")
    print(" " * 2, end="")
    print("#" * brick)


main()